xapian-glib/xapian-database.cc

```diff
@@ -808,3 +810,25 @@ xapian_database_get_flags (XapianDatabase *self)
 
   return db_flags;
 }
+
+/**
+ * xapian_database_allterms:
+ * @self: A #XapianDatabase
+ * @prefix: (nullable): Prefix to iterate over
+ *
+ * Start iterating over all terms in the database, optionally restricted to
+ * terms with the specified prefix.
+ *
+ * Returns: (transfer full): a #XapianTermIterator instance
+ *
+ * Since: 1.4
+ */
+XapianTermIterator *
+xapian_database_allterms (XapianDatabase *self,
```

Comment:
I missed this in the initial review.

Would it be possible to rename this to `enumerate_terms()`? The `allterms()` is more in line with the original Xapian API, but it's fairly out of place with any GLib-based projects.